src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+"""
+How to run:
+
+1. git clone https://github.com/stella-z3g/RubricEval -b public_release
+2. cd RubricEval/ && pip install -e .
+3. python scripts/convert_dataset_to_rubriceval_input_format.py  # This will generate an example instructions.json
+4. rubric_eval get_rubrics --input_path=instructions.json --output_path=instructions_with_rubrics.json
+5. rubric_eval get_completions --model_configs="['gpt-4o-2024-05-13']" --input_path=instructions_with_rubrics.json --output_path_prefix=output_
+6. rubric_eval --input_path_prefix=output_ --output_path_prefix=eval_output_
+"""
+
+
+import sys
+import os
+import ast
+from pathlib import Path
+import pandas as pd
+from typing import List
+from alpaca_eval import utils as ae_utils
+
+import fire
+
+from .helper import get_detailed_rubrics, get_model_completions, get_evaluations
+from .annotators import RubricBrainstormer
+
+CUR_DIR = Path(__file__).parent
+
+__all__ = ["get_rubrics", "get_outputs", "eval_outputs"]
+
+
+def check_df_fields(df, required_fields, optional_fields=set()):
```

Comment:
for later: let's add docstrings and Types for all functions.

Minor: shouldn't use mutable as default args. Use `optional_fields=None` and `optional_fields = optional_fields or set()`

Reply:
note to self: also need to add doctest (example code in docstring)